cpp/src/slice2matlab/Main.cpp

```diff
@@ -1181,102 +873,128 @@ namespace
         //
         out << "% " << n << "   Summary of " << n;
 
-        if (!doc.overview.empty())
+        StringList docOverview = doc->overview();
+        if (!docOverview.empty())
         {
             out << nl << "%";
-            writeDocLines(out, doc.overview, true);
+            writeDocLines(out, docOverview, true);
         }
 
         if (EnumPtr en = dynamic_pointer_cast<Enum>(p))
         {
-            out << nl << "%";
-            out << nl << "% " << n << " Properties:";
-            const EnumeratorList el = en->enumerators();
-            for (EnumeratorList::const_iterator q = el.begin(); q != el.end(); ++q)
+            const EnumeratorList enumerators = en->enumerators();
+            if (!enumerators.empty())
             {
-                StringList sl = splitComment((*q)->comment());
-                out << nl << "%   " << fixEnumerator((*q)->name());
-                if (!sl.empty())
+                out << nl << "%";
+                out << nl << "% " << n << " Properties:";
+                for (const auto& enumerator : enumerators)
                 {
-                    out << " - ";
-                    writeDocSentence(out, sl);
+                    out << nl << "%   " << fixEnumerator(enumerator->name());
+                    CommentPtr enumeratorDoc = enumerator->parseComment(matlabLinkFormatter, true);
+                    if (enumeratorDoc)
+                    {
+                        StringList enumeratorOverview = enumeratorDoc->overview();
+                        if (!enumeratorOverview.empty())
+                        {
+                            out << " - ";
+                            writeDocSentence(out, enumeratorOverview);
+                        }
+                    }
                 }
             }
         }
         else if (StructPtr st = dynamic_pointer_cast<Struct>(p))
         {
-            out << nl << "%";
-            out << nl << "% " << n << " Properties:";
-            const DataMemberList dml = st->dataMembers();
-            for (DataMemberList::const_iterator q = dml.begin(); q != dml.end(); ++q)
+            const DataMemberList members = st->dataMembers();
+            if (!members.empty())
             {
-                StringList sl = splitComment((*q)->comment());
-                out << nl << "%   " << fixIdent((*q)->name());
-                if (!sl.empty())
+                out << nl << "%";
+                out << nl << "% " << n << " Properties:";
+                for (const auto& member : members)
                 {
-                    out << " - ";
-                    writeDocSentence(out, sl);
+                    out << nl << "%   " << fixIdent(member->name());
+                    CommentPtr memberDoc = member->parseComment(matlabLinkFormatter, true);
+                    if (memberDoc)
```

Comment:
I think it is preferable to declare this inside the `if` with modern C++.
```suggestion
                    if (CommentPtr memberDoc = member->parseComment(matlabLinkFormatter, true))
```